test/unit/test_macros.py

```diff
@@ -0,0 +1,123 @@
+        return self.jinja_env.get_template(template_filename, globals=self.default_context)
+
+
+    def __run_macro(self, template, name, *args):
+        value = getattr(template.module, name)(*args)
+        return re.sub(r'\s\s+', ' ', value)
+ 
+ 
+    def test_macros_load(self):
+        self.jinja_env.get_template('adapters.sql')
+ 
+ 
+    def test_macros_create_table_as(self):
+        template = self.__get_template('adapters.sql')
+
+        self.assertEqual(self.__run_macro(template, 'spark__create_table_as', False, 'my_table', 'select 1'),
+                         "create table my_table as select 1")
+
+
+    def test_macros_create_table_as_file_format(self):
+        template = self.__get_template('adapters.sql')
+
+
+        self.config['file_format'] = 'delta'
+        self.assertEqual(self.__run_macro(template, 'spark__create_table_as', False, 'my_table', 'select 1'),
+                         "create table my_table using delta as select 1")
+
+
+    def test_macros_create_table_as_partition(self):
+        template = self.__get_template('adapters.sql')
+
+
+        self.config['partition_by'] = 'partition_1'
+        self.assertEqual(self.__run_macro(template, 'spark__create_table_as', False, 'my_table', 'select 1'),
+                         "create table my_table partitioned by (partition_1) as select 1")
+
+
+    def test_macros_create_table_as_partitions(self):
+        template = self.__get_template('adapters.sql')
+
+
+        self.config['partition_by'] = ['partition_1', 'partition_2']
+        self.assertEqual(self.__run_macro(template, 'spark__create_table_as', False, 'my_table', 'select 1'),
+                         "create table my_table partitioned by (partition_1,partition_2) as select 1")
+
+
+    def test_macros_create_table_as_cluster(self):
+        template = self.__get_template('adapters.sql')
+
+
+        self.config['clustered_by'] = 'cluster_1'
+        self.config['buckets'] = '1'
+        self.assertEqual(self.__run_macro(template, 'spark__create_table_as', False, 'my_table', 'select 1'),
+                         "create table my_table clustered by (cluster_1) into 1 buckets as select 1")
+
+
+    def test_macros_create_table_as_clusters(self):
+        template = self.__get_template('adapters.sql')
+
+
+        self.config['clustered_by'] = ['cluster_1', 'cluster_2']
+        self.config['buckets'] = '1'
+        self.assertEqual(self.__run_macro(template, 'spark__create_table_as', False, 'my_table', 'select 1'),
+                         "create table my_table clustered by (cluster_1,cluster_2) into 1 buckets as select 1")
+
+
+    def test_macros_create_table_as_location(self):
+        template = self.__get_template('adapters.sql')
+
+
+        self.config['location'] = '/mnt/root'
+        self.assertEqual(self.__run_macro(template, 'spark__create_table_as', False, 'my_table', 'select 1'),
+                         "create table my_table location '/mnt/root' as select 1")
```

Comment:
Possible additional test (if we can get it working):

```python
    def test_macros_create_table_as_location_dynamic(self):
        template = self.__get_template('adapters.sql')
        self.config['location'] = '/mnt/root/{{ this.table }}'
        self.assertEqual(self.__run_macro(template, 'spark__create_table_as', False, 'my_table', 'select 1'),
                         "create table my_table location '/mnt/root/my_table' as select 1")
```

Reply:
@jtcohen6 - For this proposed test (modeled after the one there in the test file), we are editing directly the dictionary `config` object, which I believe corresponds to behavior as if set within yaml. For that scenario, would the above be a valid test? (Apologies: I don't have a valid test environment on my machine yet.)

Reply:
Hi Aaron, I'm mocking most stuff normally provided by dbt, but I'm not mocking this. Maybe @jtcohen6 has an idea how to be able to create a jinja template environment dbt is using internally. Then, stuff like this would be possible to test. Otherwise, I can add a mock this which might mimick the same behaviour